src/basic/strv.h

```diff
@@ -96,10 +96,21 @@ bool strv_overlap(char **a, char **b) _pure_;
 #define STRV_FOREACH(s, l)                      \
         for ((s) = (l); (s) && *(s); (s)++)
 
-#define STRV_FOREACH_BACKWARDS(s, l)            \
-        STRV_FOREACH(s, l)                      \
-                ;                               \
-        for ((s)--; (l) && ((s) >= (l)); (s)--)
+#define STRV_FOREACH_BACKWARDS(s, l)                                \
+        for (s = ({                                                 \
+                        char **_l = l;                              \
+                        if (l) {                                    \
+                                unsigned _c;                        \
+                                _c = strv_length(_l);               \
+                                if (_c <= 0)                        \
```

Comment:
If _l == NULL, strv_length(l) == 0. So this whole condition can be removed, to simply return `l + max(strv_length(l),1)-1` ?